framework/src/modules/chain/synchronizer/block_synchronization_mechanism.js

```diff
@@ -1,14 +1,116 @@
+const { maxBy, groupBy } = require('lodash');
+const ForkChoiceRule = require('../blocks/fork_choice_rule');
+
 class BlockSynchronizationMechanism {
-	constructor() {
+	constructor({ channel, modules: { blocks } }) {
+		this.channel = channel;
 		this.active = false;
+		this.modules = {
+			blocks,
+		};
 	}
 
-	// eslint-disable-next-line class-methods-use-this,no-empty-function
-	async run() {}
+	async run() {
+		this.active = true;
+		const { connectedPeers: peers } = await this.channel.invoke(
+			'network:getNetworkStatus'
+		);
+		if (!peers.length) throw new Error('Connected peers list is empty');
+		// eslint-disable-next-line no-unused-vars
+		const bestPeer = this._computeBestPeer(peers);
+		// TODO: handle bestPeer and move on to step 2 defined in
+		// https://github.com/LiskHQ/lips/blob/master/proposals/lip-0014.md#block-synchronization-mechanism
+		// ...
+		this.active = false;
```

Comment:
if it throws above, it will be active forever 😢

Reply:
This function will be completed when all the steps in BSM are implemented separately. I will keep this in mind. Thanks